exchanges/cryptodotcom/cryptodotcom.go

```diff
@@ -0,0 +1,816 @@
+	// cryptodotcom websocket endpoints.
+	cryptodotcomWebsocketUserAPI   = "wss://stream.crypto.com/v2/user"
+	cryptodotcomWebsocketMarketAPI = "wss://stream.crypto.com/v2/market"
+
+	cryptodotcomAPIVersion = "/v2/"
+
+	// Public endpoints
+	publicAuth        = "public/auth"
+	publicInstruments = "public/get-instruments"
+	publicOrderbook   = "public/get-book"
+	publicCandlestick = "public/get-candlestick"
+	publicTicker      = "public/get-ticker"
+	publicTrades      = "public/get-trades"
+
+	// Authenticated endpoints
+	privateSetCancelOnDisconnect = "private/set-cancel-on-disconnect"
+	privateGetCancelOnDisconnect = "private/get-cancel-on-disconnect"
+
+	privateCreateSubAccountTransfer = "private/create-sub-account-transfer"
+
+	privateCreateOrder     = "private/create-order"
+	privateCancelOrder     = "private/cancel-order"
+	privateCreateOrderList = "private/create-order-list"
+	privateCancelOrderList = "private/cancel-order-list"
+	privateCancelAllOrders = "private/cancel-all-orders"
+	privateGetOrderHistory = "private/get-order-history"
+	privateGetOpenOrders   = "private/get-open-orders"
+	privateGetOrderDetail  = "private/get-order-detail"
+	privateGetTrades       = "private/get-trades"
+	privateGetTransactions = "private/get-transactions"
+
+	// Wallet management API
+	privateWithdrawal = "private/create-withdrawal"
+
+	privateGetCurrencyNetworks = "private/get-currency-networks"
+	privateGetDepositAddress   = "private/get-deposit-address"
+	privateGetAccounts         = "private/get-accounts"
+
+	// OTC Trading API
+	privateGetOTCUser         = "private/otc/get-otc-user"
+	privateGetOTCInstruments  = "private/otc/get-instruments"
+	privateOTCRequestQuote    = "private/otc/request-quote"
+	privateOTCAcceptQuote     = "private/otc/accept-quote"
+	privateGetOTCQuoteHistory = "private/otc/get-quote-history"
+	privateGetOTCTradeHistory = "private/otc/get-trade-history"
+
+	privateGetWithdrawalHistory = "private/get-withdrawal-history"
+	privateGetDepositHistory    = "private/get-deposit-history"
+
+	// Spot Trading API
+	privateGetAccountSummary = "private/get-account-summary"
+)
+
+// GetInstruments provides information on all supported instruments
+func (cr *Cryptodotcom) GetInstruments(ctx context.Context) ([]Instrument, error) {
+	var resp *InstrumentList
+	return resp.Instruments, cr.SendHTTPRequest(ctx, exchange.RestSpot, publicInstruments, request.Unset, &resp)
+}
+
+// GetOrderbook retches the public order book for a particular instrument and depth.
+func (cr *Cryptodotcom) GetOrderbook(ctx context.Context, instrumentName string, depth int64) (*OrderbookDetail, error) {
+	params := url.Values{}
+	if instrumentName == "" {
+		return nil, errSymbolIsRequired
+	}
+	params.Set("instrument_name", instrumentName)
+	if depth != 0 {
+		params.Set("depth", strconv.FormatInt(depth, 10))
+	}
+	var resp *OrderbookDetail
+	return resp, cr.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues(publicOrderbook, params), request.Unset, &resp)
```

Comment:
You are not setting the ratelimits. I understand that this exchange has different style of ratelimits, but it seems you have defined them all but don't use them